src/pkg/packager/lint/lint.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package lint contains functions for verifying zarf yaml files are valid
+package lint
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/defenseunicorns/zarf/src/pkg/layout"
+	"github.com/defenseunicorns/zarf/src/pkg/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+const (
+	validatorInvalidPrefix = "schema is invalid:"
+	validatorWarningPrefix = "zarf schema warning:"
+)
+
+// ZarfSchema is exported so main.go can embed the schema file
+var ZarfSchema embed.FS
+
+func getSchemaFile() ([]byte, error) {
+	return ZarfSchema.ReadFile("zarf.schema.json")
+}
+
+// ValidateZarfSchema validates a zarf file against the zarf schema, returns *validator with warnings or errors if they exist
+// along with an error if the validation itself failed
+func ValidateZarfSchema(path string) (*Validator, error) {
+	validator := Validator{}
+	var err error
+	if err := utils.ReadYaml(filepath.Join(path, layout.ZarfYAML), &validator.typedZarfPackage); err != nil {
+		return &validator, err
```

Comment:
We should probably return `nil` instead of returning a likely invalid pointer value of `&validator` in the case of an error

```suggestion
		return nil, err
```